src/kvstore/NebulaStore.cpp

```diff
@@ -359,14 +359,31 @@ ErrorOr<nebula::cpp2::ErrorCode, HostAddr> NebulaStore::partLeader(GraphSpaceID
 void NebulaStore::addSpace(GraphSpaceID spaceId, bool isListener) {
   folly::RWSpinLock::WriteHolder wh(&lock_);
   if (!isListener) {
+    // Iterate over all engines to ensure that each dataPath has an engine
     if (this->spaces_.find(spaceId) != this->spaces_.end()) {
       LOG(INFO) << "Data space " << spaceId << " has existed!";
-      return;
-    }
-    LOG(INFO) << "Create data space " << spaceId;
-    this->spaces_[spaceId] = std::make_unique<SpacePartInfo>();
-    for (auto& path : options_.dataPaths_) {
-      this->spaces_[spaceId]->engines_.emplace_back(newEngine(spaceId, path, options_.walPath_));
+      for (auto& path : options_.dataPaths_) {
```

Comment:
Why not build the engine in `loadPartFromPartManager`, `addSpace` is called for each partition in meta, which is duplicated.

Reply:
Would you add a test case if possible.

Reply:
First, `addSpace` is processed once per spaceId
Secondly, the `addSpace` interface is to create an engine for each dataPath of the current host, check and create an engine according to the dataPath and put it in addSpace to feel more unified.
wdyt?

Reply:
I got it, my bad.